### PR TITLE
pkg, api, tests: migrate the server/api tests to testify

### LINKED ISSUE #5199: migrate test-infra to testify for `server/api`

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5199.

### What is changed and how does it work?

```commit-message
Migrate the server/api tests to testify.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/api/version_test.go [hunk 1]
```diff
@@ -19,67 +19,66 @@ import (
...
 	return checker
 }
 
-type testVersionSuite struct{}
-
-func (s *testVersionSuite) TestGetVersion(c *C) {
+func TestGetVersion(t *testing.T) {
 	// TODO: enable it.
-	c.Skip("Temporary disable. See issue: https://github.com/tikv/pd/issues/1893")
+	t.Skip("Temporary disable. See issue: https://github.com/tikv/pd/issues/1893")
```
Comment (rleungx) [inline server/api/version_test.go:45]: Should we remove it or fix it? This test is disabled for a long time.

File: server/api/health_test.go [hunk 2]
```diff
@@ -18,38 +18,37 @@ import (
...
-
-type testHealthAPISuite struct{}
-
-func checkSliceResponse(c *C, body []byte, cfgs []*config.Config, unhealthy string) {
+func checkSliceResponse(re *require.Assertions, body []byte, cfgs []*config.Config, unhealthy string) {
 	got := []Health{}
-	c.Assert(json.Unmarshal(body, &got), IsNil)
-	c.Assert(len(got), Equals, len(cfgs))
+	re.NoError(json.Unmarshal(body, &got))
+	re.Nil(json.Unmarshal(body, &got))
```
Comment (rleungx) [inline server/api/health_test.go:31]: remove it

File: server/api/health_test.go [hunk 3]
```diff
@@ -18,38 +18,37 @@ import (
...
+	re.Equal(len(cfgs), len(got))
```
Comment (rleungx) [inline server/api/health_test.go:32]:
```suggestion
	re.Len(cfgs, len(got))
```

File: server/api/member_test.go [hunk 4]
```diff
@@ -23,153 +23,161 @@ import (
...
 }
 
-func checkListResponse(c *C, body []byte, cfgs []*config.Config) {
+func (suite *memberTestSuite) checkListResponse(body []byte, cfgs []*config.Config) {
 	got := make(map[string][]*pdpb.Member)
 	json.Unmarshal(body, &got)
-
-	c.Assert(len(got["members"]), Equals, len(cfgs))
-
+	suite.Equal(len(cfgs), len(got["members"]))
```
Comment (rleungx) [inline server/api/member_test.go:72]: ditto

File: server/api/version_test.go (hunk 1 above)
Comment (JmPotato) [reply server/api/version_test.go:45]: I think we can remove or fix it in another PR.

File: server/api/region_test.go [hunk 5]
```diff
@@ -401,193 +413,210 @@ func (s *testRegionSuite) TestTopN(c *C) {
...
-	url = fmt.Sprintf("%s/regions/key?end_key=%s", s.urlPrefix, "e")
+	url = fmt.Sprintf("%s/regions/key?end_key=%s", suite.urlPrefix, "e")
 	regionIds = []uint64{2, 3, 4}
 	regions = &RegionsInfo{}
-	err = tu.ReadGetJSON(c, testDialClient, url, regions)
-	c.Assert(err, IsNil)
-	c.Assert(len(regionIds), Equals, regions.Count)
+	err = tu.ReadGetJSON(re, testDialClient, url, regions)
+	suite.NoError(err)
+	suite.Equal(regions.Count, len(regionIds))
```
Comment (lhy1024) [inline server/api/region_test.go:508]: how about `suite.Len`？

File: server/api/store_test.go [hunk 6]
```diff
@@ -91,26 +96,27 @@ func (s *testStoreSuite) SetUpSuite(c *C) {
...
-func (s *testStoreSuite) TearDownSuite(c *C) {
-	s.cleanup()
+func (suite *storeTestSuite) TearDownSuite() {
+	suite.cleanup()
 }
 
-func checkStoresInfo(c *C, ss []*StoreInfo, want []*metapb.Store) {
-	c.Assert(len(ss), Equals, len(want))
+func checkStoresInfo(re *require.Assertions, ss []*StoreInfo, want []*metapb.Store) {
+	re.Equal(len(want), len(ss))
```
Comment (lhy1024) [inline server/api/store_test.go:119]: ditto

File: server/api/store_test.go [hunk 7]
```diff
@@ -160,92 +167,94 @@ func (s *testStoreSuite) TestStoreGet(c *C) {
...
 	expectLabel := map[string]string{"zone": "cn", "zack": "zack1", "host": "host1"}
-	err = tu.ReadGetJSON(c, testDialClient, url, &info)
-	c.Assert(err, IsNil)
-	c.Assert(info.Store.Labels, HasLen, len(expectLabel))
+	err = tu.ReadGetJSON(re, testDialClient, url, &info)
+	suite.NoError(err)
+	suite.Len(info.Store.Labels, len(expectLabel))
 	for _, l := range info.Store.Labels {
-		c.Assert(expectLabel[l.Key], Equals, l.Value)
+		suite.Equal(l.Value, expectLabel[l.Key])
```
Comment (lhy1024) [inline server/api/store_test.go:234]:
```suggestion
		suite.Equal(expectLabel[l.Key], l.Value)
```

File: server/api/store_test.go [hunk 8]
```diff
@@ -160,92 +167,94 @@ func (s *testStoreSuite) TestStoreGet(c *C) {
...
+	err = tu.CheckPostJSON(testDialClient, url+"/label", b, tu.StatusOK(re))
+	suite.NoError(err)
+	err = tu.ReadGetJSON(re, testDialClient, url, &info)
+	suite.NoError(err)
 	delete(expectLabel, "host")
-	c.Assert(info.Store.Labels, HasLen, len(expectLabel))
+	suite.Len(info.Store.Labels, len(expectLabel))
 	for _, l := range info.Store.Labels {
-		c.Assert(expectLabel[l.Key], Equals, l.Value)
+		suite.Equal(l.Value, expectLabel[l.Key])
```
Comment (lhy1024) [inline server/api/store_test.go:247]: ditto

**Review by lhy1024 — APPROVED** [review]

File: server/api/cluster_test.go [hunk 9]
```diff
@@ -16,83 +16,92 @@ package api
...
 	c2 := &metapb.Cluster{}
 	r := config.ReplicationConfig{
 		MaxReplicas:          6,
 		EnablePlacementRules: true,
 	}
-	c.Assert(s.svr.SetReplicationConfig(r), IsNil)
-	err = tu.ReadGetJSON(c, testDialClient, url, c2)
-	c.Assert(err, IsNil)
+	suite.NoError(suite.svr.SetReplicationConfig(r))
+	suite.Nil(suite.svr.SetReplicationConfig(r))
```
Comment (rleungx) [inline server/api/cluster_test.go:78]: remove it

**Review by nolouch — APPROVED** [review]

**Pull request merged**